src/core/services/compiler/compiler.js

```diff
@@ -444,27 +452,33 @@ function MdCompilerProvider($compileProvider) {
 
   /**
    * Creates and instantiates a new controller with the specified options.
-   * @param {!Object} options Options that include the controller
+   * @param {!Object} options Options that include the controller function or string
    * @param {!Object} injectLocals Locals to to be provided in the controller DI.
    * @param {!Object} locals Locals to be injected to the controller.
    * @returns {!Object} Created controller instance.
    */
   MdCompilerService.prototype._createController = function(options, injectLocals, locals) {
-    // The third and fourth arguments to $controller are considered private and are undocumented:
-    // https://github.com/angular/angular.js/blob/master/src/ng/controller.js#L86
-    // Passing `true` as the third argument causes `$controller` to return a function that
-    // gets the controller instance instead returning of the instance directly. When the
-    // controller is defined as a function, `invokeCtrl.instance` is the *same instance* as
-    // `invokeCtrl()`. However, then the controller is an ES6 class, `invokeCtrl.instance` is a
-    // *different instance* from `invokeCtrl()`.
-    var invokeCtrl = this.$controller(options.controller, injectLocals, true, options.controllerAs);
-
-    if (getPreAssignBindingsEnabled() && options.bindToController) {
-      angular.extend(invokeCtrl.instance, locals);
-    }
+    var ctrl;
+    if (angular.version.major === 1 && angular.version.minor === 7) {
```

Comment:
`=== 7` -> `>= 7`.

Reply:
Unfortunately, you need to check for 1.7.1 here. In 1.7.0 the `options.controllerAs` should be the 4th argument 😒
(It is still OK if you don't support pre-assigning in 1.7.0 so that material bahaves consistently with all 1.7+ versions, but you need to call `$controller` differently on 1.7.0 vs 1.7.1.)

Reply:
`controllerAs` is an internal parameter, isn't it? Is there a way to stop relying on it?

Reply:
It should be possible. It is only used for assigning the controller instance onto `locals.$scope`. It could be done manually with `injectLocals.$scope[options.controllerAs] = <instance>`.

Reply:
That seems like a better way to do it.